dbms/src/Functions/FunctionsJSON.cpp

```diff
@@ -0,0 +1,361 @@
+#include <Functions/FunctionsJSON.h>
+
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionFactory.h>
+
+
+namespace DB
+{
+template <typename T>
+class JSONNullableImplBase
+{
+public:
+    static DataTypePtr getType() { return std::make_shared<DataTypeNullable>(std::make_shared<T>()); }
+
+#ifdef __AVX2__
+    static Field getDefault() { return {}; }
+#endif
+};
+
+class JSONHasImpl : public JSONNullableImplBase<DataTypeUInt8>
+{
+public:
+    static constexpr auto name{"jsonHas"};
+
+#ifdef __AVX2__
+    static Field getValue(ParsedJson::iterator &) { return {1}; }
+#endif
+};
+
+class JSONLengthImpl : public JSONNullableImplBase<DataTypeUInt64>
+{
+public:
+    static constexpr auto name{"jsonLength"};
+
+#ifdef __AVX2__
+    static Field getValue(ParsedJson::iterator & pjh)
+    {
+        if (!pjh.is_object_or_array())
+            return getDefault();
+
+        size_t size = 0;
+
+        if (pjh.down())
+        {
+            size += 1;
+
+            while (pjh.next())
+                size += 1;
+        }
+
+        return {size};
+    }
+#endif
+};
+
+class JSONTypeImpl : public JSONNullableImplBase<DataTypeString>
+{
+public:
+    static constexpr auto name{"jsonType"};
+
+#ifdef __AVX2__
+    static Field getValue(ParsedJson::iterator & pjh)
+    {
+        switch (pjh.get_type())
+        {
+            case '[':
+                return "Array";
+            case '{':
+                return "Object";
+            case '"':
+                return "String";
+            case 'l':
+                return "Int64";
+            case 'd':
+                return "Float64";
+            case 't':
+                return "Bool";
+            case 'f':
+                return "Bool";
+            case 'n':
+                return "Nothing";
```

Comment:
Why do you name it "Nothing"? It's `null` literal, isn't it?

Reply:
Make sense